pkg/apis/cluster/v1alpha1/types.go

```diff
@@ -91,3 +86,29 @@ type ClusterList struct {
 
 	Items []Cluster `json:"items"`
 }
+
+// Conditions and ConditionReasons for the kcp Cluster object.
+const (
+	// ClusterReadyCondition means the Cluster is available.
+	ClusterReadyCondition conditionsv1alpha1.ConditionType = "Ready"
+
+	// ClusterUnknownReason documents a Cluster which readyness is unknown.
+	ClusterUnknownReason = "ClusterStatusUnknown"
+
+	// ClusterReadyReason documents a Cluster that is ready.
+	ClusterReadyReason = "ClusterReady"
+
+	// ClusterNotReadyReason documents a Cluster is not ready, when the "readyz" check returns false.
+	ClusterNotReadyReason = "ClusterNotReady"
```

Comment:
I don't think a constant reason is the intention of the reason field. This should basically classify your error message. E.g. SyncerInstallFailed, SyncerNotReady (this only one of many), SyncerStartFailed

Reply:
I was updating those conditions thinking on the "next steps", the agent-based approach, where the syncer will update those fields. But let me add the current cases. We can simplify it later!

Reply:
Updated! Please check again.